FeedStoreChallenge/CoreData Feed Store/CoreDataFeedStore.swift

```
@@ -0,0 +1,99 @@
+//
+//  CoreDataFeedStore.swift
+//  FeedStoreChallenge
+//
+//  Created by Nicolas De Maio on 9/20/20.
+//  Copyright © 2020 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+public class CoreDataFeedStore: FeedStore {
+    private let modelName: String
+    private let bundle: Bundle
+    
+    public init(model: String = "FeedStore", bundle: Bundle = Bundle(for: CoreDataFeedStore.self)) {
+        modelName = model
+        self.bundle = bundle
+    }
+    
+    lazy var persistentContainer: NSPersistentContainer? = {
+        if let mom = NSManagedObjectModel.mergedModel(from: [bundle]) {
+            let container = NSPersistentContainer(name: modelName, managedObjectModel: mom)
+                    
+            container.loadPersistentStores(completionHandler: { (storeDescription, error) in
+                if let error = error as NSError? {
+                    fatalError("Unresolved error \(error), \(error.userInfo)")
```

It'd be best to handle the possible error. For example, you could throw the error, so the clients know something went wrong. 

Moreover, you shouldn't be able to instantiate a `CoreDataFeedStore` with a wrong model name or bundle. So you could run those checks in the initializer. If it fails to find the `NSManagedObjectModel` or create the `NSPersistentContainer`, it should throw.

So you never have a `CoreDataFeedStore` instance that doesn't work (e.g., it can't work without an NSPersistentContainer!)